### PR TITLE
 ios-sdk-pkgs: Init from iOS SDK from XCode

### PR DESCRIPTION
**Pull request by kmicklas**
###### Motivation for this change

This is basically a rewrite of the existing iOS support to take advantage of the new cross compilation infrastructure.

Limitations/issues:

 * Currently the SDK is used impurely but I want to eventually have the user import the SDK into the store and use it purely.
 * SDK versions newer than 10.3 are not supported. These will require a newer `cctools` which has it's `.tbd` linking support split out into `libtapi` which seems hard to build.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

CC @ElvishJerricco @shlevy

### CODE REVIEW

**Comment by Ericson2314** [comment]
Perfect!

**Review by Ericson2314 — APPROVED** [review]
Perfect!

**Comment by ElvishJerricco** [comment]
Is there an example for using this anywhere?

**Comment by Ericson2314** [comment]
Ah I should add/change some `lib/systems/examples.nix` entries.

File: pkgs/os-specific/darwin/ios-cross/default.nix [hunk 1]
```diff
@@ -1,61 +0,0 @@
...
-, binutils
-, stdenv
-, coreutils
-, gnugrep
-, buildPackages
-, hostPlatform
-, targetPlatform
-}:
-
-/* As of this writing, known-good prefix/arch/simulator triples:
```
Comment (Ericson2314) [inline pkgs/os-specific/darwin/ios-cross/default.nix]: @elvishjerricco can you update `lib/systems/examples.nix` based on this? I tried to do it on my phone but GitHub makes it too hard.

Comment (Ericson2314) [reply pkgs/os-specific/darwin/ios-cross/default.nix]: All 4 should have `useiOSPrebuilt = false;`

Comment (ElvishJerricco) [reply pkgs/os-specific/darwin/ios-cross/default.nix]: I'm not actually sure what `crossSystem` is supposed to look like for this.

**Pull request merged**